cmd/sti/main.go

```diff
@@ -100,57 +98,57 @@ func newCmdBuild(req *api.Request) *cobra.Command {
 			}()
 			// Attempt to restore the build command from the configuration file
 			if useConfig {
-				config.Restore(req, cmd)
+				config.Restore(cfg, cmd)
 			}
 
 			// If user specifies the arguments, then we override the stored ones
 			if len(args) >= 2 {
-				req.Source = args[0]
-				req.BaseImage = args[1]
+				cfg.Source = args[0]
+				cfg.BuilderImage = args[1]
 				if len(args) >= 3 {
-					req.Tag = args[2]
+					cfg.Tag = args[2]
 				}
 			}
 
-			if !validRequest(req) {
+			if len(validation.ValidateConfig(cfg)) != 0 {
```

Comment:
I think it'd be ideal if we moved some of this logic to a function in a different package (anything that doesn't have to do with the environment or command line args) so we can reuse it from origin. Otherwise we'd have to repeat the same code in origin (validating the config, creating the build strategy, etc)